src/control/server/storage/metadata/provider.go

```diff
@@ -144,15 +143,28 @@ func (p *Provider) setupRootDir(req storage.MetadataFormatRequest) error {
 	return nil
 }
 
-func isUsableFS(fs string) bool {
-	switch fs {
+func (p *Provider) isUsableFS(fs *system.FsType, path string) bool {
+	if fs.NoSUID {
+		p.log.Errorf("cannot use filesystem with nosuid flag set (%s) for control metadata", path)
+		return false
+	}
+
+	switch fs.Name {
 	case system.FsTypeNfs:
 		// NB: Using NFS as a metadata location could cause conflicts if all servers have
 		// access to the location, which is intended to store node-local data.
 		// The daos_server_helper also gets into some trouble creating the directory if
 		// root squash is enabled.
+		p.log.Errorf("cannot use filesystem %s (%s) for control metadata", fs.Name, path)
```

Comment:
again a bit more explanation here would be useful

Reply:
Added some more detail to the comment